### PR TITLE
Added node.js support doc

### PR DESCRIPTION
**Pull request by delvedor**
Since the client **must** follow the stack versioning, it's really hard to follow the [node.js release cadence](https://nodejs.org/en/about/releases/), so we have decided to allow dropping EOL node.js version in minor releases. This pr contains the formalization of the process we will follow in the future.

- [x] asciidoc updated
- [x] README updated

### CODE REVIEW

**Review by philkra — APPROVED** [review]
LGTM

File: docs/installation.asciidoc [hunk 1]
```diff
@@ -20,13 +20,48 @@ npm install @elastic/elasticsearch@<major>
 To learn more about the supported major versions, please refer to the 
 <<js-compatibility-matrix>>.
 
+[discrete]
+[[nodejs-support]]
+=== Node.js support
+
+NOTE: The minimum supported version of Node.js is `v8`.
+
+The client versioning follows the Elastic Stack versioning, this means that
```
Comment (szabosteve) [inline docs/installation.asciidoc:29]:
```suggestion
The client versioning follows the {stack} versioning, this means that
```

File: docs/installation.asciidoc [hunk 2]
```diff
@@ -20,13 +20,48 @@ npm install @elastic/elasticsearch@<major>
...
+major, minor, and patch releases are done following a precise schedule that
+often does not coincide with the https://nodejs.org/en/about/releases/[Node.js release] times.
+
+For avoiding supporting insecure and unsupported versions of Node.js, the
```
Comment (szabosteve) [inline docs/installation.asciidoc:33]:
```suggestion
To avoid support issues and unsupported versions of Node.js, the
```

File: docs/installation.asciidoc [hunk 3]
```diff
@@ -20,13 +20,48 @@ npm install @elastic/elasticsearch@<major>
...
+client *will drop the support of EOL versions of Node.js between minor releases*.
+Typically, as soon as a Node.js version goes into EOL, the client will continue
+to support that version for at least another minor release. If you are using the client
+with a version of Node.js that will be unsupported soon, you will see a warning
+in your logs (the client will start logging the warning with two minors in advance).
+
+Because of the reasons defined above, unless you are *always* using a supported
+version of Node.js, we recommend defining the client dependency in your
+`package.json` with the `~` instead of `^`. In this way, you will lock the
+dependency on the minor release and not the major. (eg: `~7.10.0` instead
```
Comment (szabosteve) [inline docs/installation.asciidoc:43]:
```suggestion
dependency on the minor release and not the major. (for example, `~7.10.0` instead
```

File: docs/installation.asciidoc [hunk 4]
```diff
@@ -20,13 +20,48 @@ npm install @elastic/elasticsearch@<major>
...
+
+For avoiding supporting insecure and unsupported versions of Node.js, the
+client *will drop the support of EOL versions of Node.js between minor releases*.
+Typically, as soon as a Node.js version goes into EOL, the client will continue
+to support that version for at least another minor release. If you are using the client
+with a version of Node.js that will be unsupported soon, you will see a warning
+in your logs (the client will start logging the warning with two minors in advance).
+
+Because of the reasons defined above, unless you are *always* using a supported
+version of Node.js, we recommend defining the client dependency in your
```
Comment (szabosteve) [inline docs/installation.asciidoc:41]:
```suggestion
Unless you are *always* using a supported version of Node.js, 
we recommend defining the client dependency in your
```

**Review by szabosteve — APPROVED** [review]
Thank you for adding this section @delvedor. LGTM.
I left some very minor suggestions, mostly nits. Please feel free to take or leave them.

File: docs/installation.asciidoc (hunk 2 above)
Comment (delvedor) [reply docs/installation.asciidoc:33]: I want to keep the "insecure", as using an EOL version is not safe :)

**Review by Mpdreamz — APPROVED** [review]
LGTM!
